From 624b2068eeb81c3cbf432629468fae80205f2478 Mon Sep 17 00:00:00 2001
Message-Id: <624b2068eeb81c3cbf432629468fae80205f2478.1414143271.git.chang-joon.lee@intel.com>
In-Reply-To: <f393e797ccaaf66a25ee3d7c4e6136083a6effeb.1414143271.git.chang-joon.lee@intel.com>
References: <f393e797ccaaf66a25ee3d7c4e6136083a6effeb.1414143271.git.chang-joon.lee@intel.com>
From: Yogesh Mohan Marimuthu <yogesh.mohan.marimuthu@intel.com>
Date: Thu, 16 Oct 2014 18:54:04 +0530
Subject: [PATCH 13/26] REVERTME [VPG]: drm/i915: configure dual link pclk
 only for >= CHT B0

Configure pixel overlap only for CHT stepping >= B0.

REVERTME: This patch can be reverted once everyone in Intel has
stopped using CHT SoC Ax stepping.

For: GMINL-1532
Change-Id: Ie4ec24847386ac01519b46e0a9d8b92f308153d5
Signed-off-by: Gaurav K Singh <gaurav.k.singh@intel.com>
Signed-off-by: Yogesh Mohan Marimuthu <yogesh.mohan.marimuthu@intel.com>
---
 drivers/gpu/drm/i915/intel_dsi_panel_vbt.c |    8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_dsi_panel_vbt.c b/drivers/gpu/drm/i915/intel_dsi_panel_vbt.c
index 1478ab6..c78ac58 100644
--- a/drivers/gpu/drm/i915/intel_dsi_panel_vbt.c
+++ b/drivers/gpu/drm/i915/intel_dsi_panel_vbt.c
@@ -595,7 +595,13 @@ static bool generic_init(struct intel_dsi_device *dsi)
 	if (intel_dsi->dual_link) {
 		pclk = pclk / 2;
 
-		if (intel_dsi->dual_link & MIPI_DUAL_LINK_FRONT_BACK) {
+		/*
+		 * in case of CHT B0 and above stepping we can enable
+		 * pixel_overlap if needed by panel. In this case
+		 * we need to increase the pixelclock for extra pixels
+		 */
+		if ((IS_CHERRYVIEW(dev_priv->dev) && STEP_FROM(STEP_B0)) &&
+			(intel_dsi->dual_link & MIPI_DUAL_LINK_FRONT_BACK)) {
 			pclk += DIV_ROUND_UP(mode->vtotal *
 					intel_dsi->pixel_overlap * 60, 1000);
 		}
-- 
1.7.9.5

